go/controller/internal/autogen/autogen_api_translator.go

```diff
@@ -562,3 +563,169 @@ func addModelClientToConfig(
 	(*toolConfig)["model_client"] = cfg
 	return nil
 }
+
+// createModelClientForProvider creates a model client component based on the model provider
+func createModelClientForProvider(modelConfig *v1alpha1.ModelConfig, apiKey string) (*api.Component, error) {
+	switch modelConfig.Spec.Provider {
+	case v1alpha1.ProviderAnthropic:
+		anthropicConfig := map[string]interface{}{
```

Comment:
Can you create a type like we do here? https://github.com/kagent-dev/kagent/blob/main/go/autogen/api/models.go. Generic `map[string]interface{}` are not super nice to use in `go`